Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -517,55 +516,55 @@ void HTMLInputElement::create_shadow_tree_if_needed()
     case TypeAttributeState::ImageButton:
     case TypeAttributeState::Color:
         return;
-    default:
-        break;
-    }
-
-    auto shadow_root = heap().allocate<DOM::ShadowRoot>(realm(), document(), *this, Bindings::ShadowRootMode::Closed);
-    auto initial_value = m_value;
-    auto element = DOM::create_element(document(), HTML::TagNames::div, Namespace::HTML).release_value_but_fixme_should_propagate_errors();
-    MUST(element->set_attribute(HTML::AttributeNames::style, R"~~~(
-        display: flex;
-        height: 100%;
-        align-items: center;
-        white-space: pre;
-        border: none;
-        padding: 1px 2px;
+    // FIXME: This could be better factored. Everything except the above types becomes a text input.
+    default: {
+        auto shadow_root = heap().allocate<DOM::ShadowRoot>(realm(), document(), *this, Bindings::ShadowRootMode::Closed);
```

Comment:
Why? This is worse IMO

Reply:
This is one way of ensuring this path is not taken in the case of the Color type, would it be better to return instead?

Reply:
If that's the concern, I would break these out into separate functions instead of moving large blocks of code inside this `case` label. Something like:

```c++
switch (type_state()) {
case TypeAttributeState::RadioButton:
case TypeAttributeState::Checkbox:
case TypeAttributeState::Button:
case TypeAttributeState::SubmitButton:
case TypeAttributeState::ResetButton:
case TypeAttributeState::ImageButton:
case TypeAttributeState::Color:
    create_color_input_shadow_tree();
    return;
default:
    create_text_input_shadow_tree();
    break;
}
```